### PR TITLE
TLS Bootstrap and Server Cert Rotation feature documentation

### PR DESCRIPTION
**Pull request by mikedanese**
Content change is in the second commit.

### CODE REVIEW

**Comment by zparnold** [comment]
@mikedanese When you have a moment, could I please ask you to update the docs?

File: content/en/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping.md [hunk 1]
```diff
@@ -186,18 +189,30 @@ roleRef:
...
+`kubelet-bootstrap` to `kubectl config set-credentials` and include
+`--token=<token-value>` as follows:
 
 ```
 kubectl config set-credentials kubelet-bootstrap --token=${BOOTSTRAP_TOKEN} --kubeconfig=bootstrap.kubeconfig
 ```
 
-When starting the kubelet, if the file specified by `--kubeconfig` does not exist, the bootstrap kubeconfig is used to request a client certificate from the API server. On approval of the certificate request and receipt back by the kubelet, a kubeconfig file referencing the generated key and obtained certificate is written to the path specified by `--kubeconfig`. The certificate and key file will be placed in the directory specified by `--cert-dir`.
+When starting the kubelet, if the file specified by `--kubeconfig` does not
+exist, the bootstrap kubeconfig is used to request a client certificate from the
```
Comment (awly) [inline content/en/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping.md]: mention that bootstrap kubeconfig is specified via `--bootstrap-kubeconfig` flag

File: content/en/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping.md [hunk 2]
```diff
@@ -207,25 +222,46 @@ When starting the kubelet, if the file specified by `--kubeconfig` does not exis
...
 ```
---feature-gates=RotateKubeletClientCertificate=true,RotateKubeletServerCertificate=true
+--rotate-certificates
 ```
 
-`RotateKubeletClientCertificate` causes the kubelet to rotate its client certificates by creating new CSRs as its existing
-credentials expire. `RotateKubeletServerCertificate` causes the kubelet to both request a serving certificate after
-bootstrapping its client credentials and rotate the certificate. The serving cert currently does not request DNS or IP
-SANs.
+`RotateKubeletServerCertificate` causes the kubelet to both request a serving
```
Comment (awly) [inline content/en/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping.md]: Above you removed mentions of `selfnodeserver` which is needed to make this work. Either remove serving cert mentions everywhere or re-add them above.

Also, to make his work, user has to set [ServerTLSBootstrap](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/config/types.go#L112) config field.

File: content/en/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping.md [hunk 3]
```diff
@@ -207,25 +222,46 @@ When starting the kubelet, if the file specified by `--kubeconfig` does not exis
...
 
 ## kubectl approval
-The signing controller does not immediately sign all certificate requests. Instead, it waits until they have been flagged with an
-"Approved" status by an appropriately-privileged user. This is intended to eventually be an automated process handled by an external
-approval controller, but for the alpha version of the API it can be done manually by a cluster administrator using kubectl.
-An administrator can list CSRs with `kubectl get csr` and describe one in detail with `kubectl describe csr <name>`. Before the 1.6 release there were
-[no direct approve/deny commands](https://github.com/kubernetes/kubernetes/issues/30163) so an approver had to update
-the Status field directly ([rough how-to](https://github.com/gtank/csrctl)). Later versions of Kubernetes offer `kubectl certificate approve <name>` and `kubectl certificate deny <name>` commands.
+
+The signing controller does not immediately sign all certificate requests.
```
Comment (awly) [inline content/en/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping.md]: This is a bit confusing. By default `kube-controller-manager` will auto-approve certs that pass SAR review.
This paragraph can be mis-read as: "you must either manually approve all CSRs or write a custom controller to do that".

File: content/en/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping.md (hunk 2 above)
Comment (mikedanese) [reply content/en/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping.md]: `selfnodeserver` was removed. You need to write a custom approving controller for this to work, as stated in the note.

File: content/en/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping.md [hunk 4]
```diff
@@ -1,103 +1,121 @@
...
+`--client-ca-file=/var/lib/kubernetes/ca.pem`).
 
 ## kube-controller-manager configuration
-The API for requesting certificates adds a certificate-issuing control loop to the Kubernetes Controller Manager. This takes the form of a
-[cfssl](https://blog.cloudflare.com/introducing-cfssl/) local signer using assets on disk. Currently, all certificates issued have one year validity and a default set of key usages.
+
+The API for requesting certificates adds a certificate-issuing control loop to
+the Kubernetes Controller Manager. This takes the form of a
+[cfssl](https://blog.cloudflare.com/introducing-cfssl/) local signer using
+assets on disk. Currently, all certificates issued have one year validity and a
```
[the remark this answers is not in the document]
Comment (mikedanese) [reply content/en/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping.md]: Done below.

**Comment by zparnold** [comment]
@mikedanese Great job!
/apprive

**Comment by zparnold** [comment]
Oh my gosh, I misspelled a word...wow. Documentation fail. 😂 
/lgtm
/approve

**Pull request merged**